### PR TITLE
panos_ha module proposal review

### PR DESCRIPTION
**Pull request by patrickdaj**
Looking for some initial feedback on this module.

- It seemed like it made sense to keep all of this in one module rather than having a seperate module for HighAvailability, HA1, HA2, etc.
- Had an issue with two of the stranger settings in the API that I was added for HA A/A in that they did not get pulled into the HighAvailability object with a refreshall.  This led to changed always being flagged, so there is a bit of a hack to resolve this.

Anyhow it more or less works, just looking for some input.

### CODE REVIEW

**Comment by netdevops-se** [comment]
Nice, was just thinking of developing a panos_ha module.

**Comment by netdevops-se** [comment]
Verified that it works. However, I would like to see an option for setting device-priority though, is it possible to fix?

**Comment by patrickdaj** [comment]
> Verified that it works. However, I would like to see an option for setting device-priority though, is it possible to fix?

Thanks for testing.  Unfortunately pandevice.ha.HighAvailability doesn't currenlty have device-priority support.  If you add priority you kind of need preemptive as well, right?  Don't think priority does much without changinge preempt from default of disabled.

I think for now I'll leave it as is, maybe request an enhancement on pandevice and then this module can be updated easily to include that.

File: library/panos_ha.py [hunk 1]
```diff
@@ -0,0 +1,443 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+#  Copyright 2018 Palo Alto Networks, Inc
```
Comment (shinmog) [inline library/panos_ha.py]: 2019 :)

**Comment by shinmog** [comment]
@patrickdaj 

Thinking about it a bit, if you wanted to do the 2nd path, I don't think having the facts module is necessary as part of this PR.

**Pull request merged**